src/runtime_src/core/pcie/linux/device_linux.cpp

```diff
@@ -683,6 +683,50 @@ struct accel_deadlock_status
   }
 };
 
+/*
+* Versal devices with PCIe interface program sc using xgq's
+* Transaction is initiated by writing to sysfs entry
+* This function throws different exceptions for 1.sysfs entry not present 2.sysfs access failure
+* Current sysfs access returns error string for both sysfs entry not present and sysfs access error
+*/
+struct program_sc
+{
+  using value_type = query::program_sc::value_type;
+  using result_type = query::program_sc::value_type;
+  static const std::string subdev;
+  static const std::string entry;
+
+  static result_type
+  get(const xrt_core::device* device, key_type key)
+  {
+    std::string err;
+    value_type value;
+    get_pcidev(device)->sysfs_get(subdev, entry, err, value, static_cast<value_type>(-1));
+    if (!err.empty())
+      throw xrt_core::query::sysfs_error(err);
+
+    return value;
+  }
+
+  static void
+  put(const xrt_core::device* device, key_type key, const boost::any& val)
+  {
+    auto pdev = get_pcidev(device);
+    //check for exsistence of sysfs node
+    std::string sysfs_path = pdev->get_sysfs_path(subdev, entry);
+    if(sysfs_path.empty()) {
+      throw xrt_core::query::not_supported(entry + " sysfs node doesnot exist");
+    }
+
+    std::string err;
+    pdev->sysfs_put(subdev, entry, err, boost::any_cast<value_type>(val));
+    if (!err.empty())
+      throw xrt_core::query::sysfs_error(err);
+  }
+};
+const std::string program_sc::subdev = "xgq_vmr";
+const std::string program_sc::entry = "program_sc";
```

Comment:
Why are these variables defined outside query? Please move it inside query structure.

Reply:
as these are static variables I have defined them outside